iroh-util/src/lib.rs

```diff
@@ -240,16 +256,69 @@ pub fn increase_fd_limit() -> std::io::Result<u64> {
 
 #[cfg(test)]
 mod tests {
+    use std::ffi::{OsStr, OsString};
+
     use serde::Deserialize;
 
     use super::*;
 
+    /// Helper struct to restore an environment variable once dropped.
+    struct EnvDropGuard {
```

Comment:
there is this nice crate by @vmx https://crates.io/crates/temp-env so one doesn't have to reimplement this every time :)